### PR TITLE
[ONNX] Modify int[] listConstruct to support tensor arg

### PR DESCRIPTION
**Pull request by shubhambhokare1**
Avoid creating unsqueeze nodes for ListConstruct Int[] ouput case with tensor inputs
For example, if the listConstruct is (tensor[2], 1, 5), avoid adding unsqueeze nodes for tensor[2]

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -5628,6 +5628,26 @@ def forward(self, x):
         self.run_test(OnesModel(), x, input_names=["x"], dynamic_axes={"x": [0, 1, 2]})
         self.run_test(OnesModel(), x, remained_onnx_input_idx=[])
 
+    @skipIfUnsupportedMinOpsetVersion(9)
+    @disableScriptTest()
```
Comment (garymm) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:5632]: please add a comment explaining why scripting is disabled

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 2]
```diff
@@ -5628,6 +5628,26 @@ def forward(self, x):
...
+    def test_zeros_with_tensorinput(self):
```
Comment (garymm) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:5633]:
```suggestion
    def test_zeros_with_tensor_input(self):
```

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 3]
```diff
@@ -5628,6 +5628,26 @@ def forward(self, x):
...
+        class Zero_(torch.nn.Module):
+            def forward(self, x):
+                return x, torch.zeros(x, 1)
+
+        x = torch.tensor([2])
+        self.run_test(Zero_(), (x, ))
+
+    @skipIfUnsupportedMinOpsetVersion(9)
+    @disableScriptTest()
+    def test_ones_with_tensorinput(self):
```
Comment (garymm) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:5643]: What's the value in adding both of these tests rather than just 1?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 4]
```diff
@@ -5628,6 +5628,26 @@ def forward(self, x):
         self.run_test(OnesModel(), x, input_names=["x"], dynamic_axes={"x": [0, 1, 2]})
         self.run_test(OnesModel(), x, remained_onnx_input_idx=[])
 
+    @skipIfUnsupportedMinOpsetVersion(9)
+    @disableScriptTest()
+    def test_zeros_with_tensorinput(self):
+        class Zero_(torch.nn.Module):
```
Comment (garymm) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:5634]:
```suggestion
        class Zero(torch.nn.Module):
```

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 5]
```diff
@@ -5628,6 +5628,26 @@ def forward(self, x):
...
+            def forward(self, x):
+                return x, torch.zeros(x, 1)
+
+        x = torch.tensor([2])
+        self.run_test(Zero_(), (x, ))
```
Comment (garymm) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:5639]:
```suggestion
        self.run_test(Zero(), (x,))
```

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 6]
```diff
@@ -5628,6 +5628,26 @@ def forward(self, x):
         self.run_test(OnesModel(), x, input_names=["x"], dynamic_axes={"x": [0, 1, 2]})
         self.run_test(OnesModel(), x, remained_onnx_input_idx=[])
 
+    @skipIfUnsupportedMinOpsetVersion(9)
+    @disableScriptTest()
+    def test_zeros_with_tensorinput(self):
+        class Zero_(torch.nn.Module):
+            def forward(self, x):
+                return x, torch.zeros(x, 1)
```
Comment (garymm) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:5636]: Why is this returning x? I don't think it's needed to trigger the bug that this is fixing.

File: torch/csrc/jit/passes/onnx/helper.cpp [hunk 7]
```diff
@@ -186,7 +186,13 @@ Node* transformToONNXConcatNode(
   for (auto* input : lc_node->inputs()) {
     auto new_input =
         need_new_input ? g->addInput()->copyMetadata(input) : input;
-
+    // Skip unsqueeze for input node if input dim is already 1 or more
```
Comment (garymm) [inline torch/csrc/jit/passes/onnx/helper.cpp:189]: I don't understand why we need unsqueeze in the other case. Could you expand the comment on line 182 to explain?

Comment (shubhambhokare1) [reply torch/csrc/jit/passes/onnx/helper.cpp]: Expanded comment

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 8]
```diff
@@ -5629,6 +5629,16 @@ def forward(self, x):
         self.run_test(OnesModel(), x, input_names=["x"], dynamic_axes={"x": [0, 1, 2]})
         self.run_test(OnesModel(), x, remained_onnx_input_idx=[])
 
+    @skipIfUnsupportedMinOpsetVersion(9)
+    @disableScriptTest()  # torch.zeros/torch.ones with size tensor of dim != 0 not torchScript compatible.
```
Comment (garymm) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:5633]:
```suggestion
    @disableScriptTest()  # torch.zeros/torch.ones with size tensor of dim != 0 not scriptable
```

File: torch/csrc/jit/passes/onnx/helper.cpp [hunk 9]
```diff
@@ -186,7 +186,16 @@ Node* transformToONNXConcatNode(
   for (auto* input : lc_node->inputs()) {
     auto new_input =
         need_new_input ? g->addInput()->copyMetadata(input) : input;
-
+    // ONNX Concat requires tensor inputs with rank > axis(=0). Certain inputs from
```
Comment (garymm) [inline torch/csrc/jit/passes/onnx/helper.cpp:189]: `rank > axis(=0)`

this looks a bit weird. What does `axis(=0)` mean?

**Review by garymm — APPROVED** [review]

File: torch/csrc/jit/passes/onnx/helper.cpp (hunk 9 above)
Comment (shubhambhokare1) [reply torch/csrc/jit/passes/onnx/helper.cpp:189]: Described it in a more understandable way
